### DISCUSSION TITLE
Should this value be initialized?

### DISCUSSION THREAD

**Discussion by jjeffroberts** [openssl/openssl#21907] [Q&A] [answered]
https://github.com/openssl/openssl/blob/master/crypto/evp/evp_enc.c#L621

Should 'soutl' be initialized, to zero for instance?

I'm working on a provider and my OSSL_FUNC_cipher_update_fn did not explicitly set *outl because no work is actually done until the _final method.

With 'soutl' uninitialized, it came into this line as garbage and resulted in a failure.

https://github.com/openssl/openssl/blob/master/crypto/evp/evp_enc.c#L657

I can work around that by setting it to zero in my _update method (or perhaps by moving the work to that method and having _final not do much) but it was not obvious why this was failing until I looked under the covers.

**Comment by paulidale** [comment, marked as answer]
Your provider should be setting *outl to zero to indicate that it isn't producing any output for that update call.
